panic: missing root module on output call

When calling "terraform output autoscaling_group_id" terraform crashes. 

2015/03/20 15:16:57 [INFO] Terraform version: 0.3.7  0bc0c03fece07c4f21ee5195743116a4d418f234+CHANGES
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: atlas = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-atlas
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: aws = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-aws
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: cloudflare = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-cloudflare
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: cloudstack = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-cloudstack
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: consul = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-consul
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: digitalocean = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-digitalocean
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: dnsimple = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-dnsimple
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: google = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-google
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: heroku = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-heroku
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: mailgun = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-mailgun
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: null = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-null
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: file = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provisioner-file
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: local-exec = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provisioner-local-exec
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: remote-exec = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provisioner-remote-exec
2015/03/20 15:16:57 Detected home directory from env var: /home/vagrant
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: atlas = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-atlas
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: aws = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-aws
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: cloudflare = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-cloudflare
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: cloudstack = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-cloudstack
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: consul = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-consul
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: digitalocean = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-digitalocean
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: dnsimple = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-dnsimple
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: google = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-google
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: heroku = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-heroku
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: mailgun = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-mailgun
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: null = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provider-null
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: file = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provisioner-file
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: local-exec = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provisioner-local-exec
2015/03/20 15:16:57 [DEBUG] Discoverd plugin: remote-exec = /home/vagrant/project/src/environments/conf/intelligence-api/z/a/terraform-provisioner-remote-exec
2015/03/20 15:16:57 Detected home directory from env var: /home/vagrant
2015/03/20 15:16:57 [DEBUG] Attempting to open CLI config file: /home/vagrant/.terraformrc
2015/03/20 15:16:57 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2015/03/20 15:16:57 waiting for all plugin processes to complete...
2015/03/20 15:16:57 Detected home directory from env var: /home/vagrant
panic: missing root module

goroutine 1 [running]:
github.com/hashicorp/terraform/terraform.(*State).RootModule(0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/state.go:107 +0xa2
github.com/hashicorp/terraform/command.(*OutputCommand).Run(0xc20803cc30, 0xc20800a020, 0x1, 0x1, 0xc208040978)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/command/output.go:41 +0x4b1
github.com/mitchellh/cli.(*CLI).Run(0xc20805e1e0, 0xc2080840c0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/mitchellh/cli/cli.go:100 +0x3ce
main.wrappedMain(0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:150 +0xb70
main.realMain(0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:82 +0x923
main.main()
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:18 +0x1f

goroutine 5 [syscall]:
os/signal.loop()
	/Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [select]:
net/http.(*Transport).getConn(0xc208066000, 0xc20803d1e0, 0x0, 0xc208066360, 0x5, 0xc20801eda0, 0x20, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:525 +0x608
net/http.(*Transport).RoundTrip(0xc208066000, 0xc20803d1e0, 0xc2080663f0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:228 +0x4d4
net/http.send(0xc20803d1e0, 0x7f416911e090, 0xc208066000, 0x88, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:219 +0x4fc
net/http.(*Client).send(0xc34ee0, 0xc20803d1e0, 0x88, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:142 +0x15b
net/http.(*Client).doFollowingRedirects(0xc34ee0, 0xc20803d1e0, 0xa9ffd8, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:367 +0xb25
net/http.(*Client).Do(0xc34ee0, 0xc20803d1e0, 0xa, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:174 +0xa4
github.com/hashicorp/go-checkpoint.Check(0xc2080b4080, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/go-checkpoint/checkpoint.go:139 +0xab6
main.runCheckpoint(0xc20801ed80)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/checkpoint.go:51 +0x497
created by main.wrappedMain
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:99 +0x3f7

goroutine 38 [runnable]:
net.(*singleflight).doCall(0xc34510, 0xc20804a660, 0xc20801eda0, 0x1c, 0xc208033620)
	/Users/mitchellh/code/3rdparty/go/src/net/singleflight.go:90
created by net.(*singleflight).DoChan
	/Users/mitchellh/code/3rdparty/go/src/net/singleflight.go:84 +0x42b

goroutine 37 [runnable]:
runtime.timerproc()
	/Users/mitchellh/code/3rdparty/go/src/runtime/time.go:151
runtime.goexit()
	/Users/mitchellh/code/3rdparty/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 36 [select]:
net.lookupIPDeadline(0xc20801eda0, 0x1c, 0xecc9e3107, 0x3125f1c7, 0xc35040, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/lookup.go:82 +0x6cb
net.resolveInternetAddr(0x98de70, 0x3, 0xc20801eda0, 0x20, 0xecc9e3107, 0x3125f1c7, 0xc35040, 0x0, 0x0, 0x0, ...)
	/Users/mitchellh/code/3rdparty/go/src/net/ipsock.go:285 +0x49b
net.resolveAddr(0x97f790, 0x4, 0x98de70, 0x3, 0xc20801eda0, 0x20, 0xecc9e3107, 0x3125f1c7, 0xc35040, 0x0, ...)
	/Users/mitchellh/code/3rdparty/go/src/net/dial.go:110 +0x378
net.(*Dialer).Dial(0xc20803f1c0, 0x98de70, 0x3, 0xc20801eda0, 0x20, 0x0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/dial.go:158 +0xf6
net.*Dialer.Dial·fm(0x98de70, 0x3, 0xc20801eda0, 0x20, 0x0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:38 +0x79
net/http.(*Transport).dial(0xc208066000, 0x98de70, 0x3, 0xc20801eda0, 0x20, 0x0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:479 +0x84
net/http.(*Transport).dialConn(0xc208066000, 0x0, 0xc208066360, 0x5, 0xc20801eda0, 0x20, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:564 +0x1678
net/http.func·019()
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:520 +0x42
created by net/http.(*Transport).getConn
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:522 +0x335

--- a/command/output.go
+++ b/command/output.go
@@ -39,7 +39,7 @@ func (c *OutputCommand) Run(args []string) int {
 	}
 
 	state := stateStore.State()
-	if len(state.RootModule().Outputs) == 0 {
+	if state.Empty() || len(state.RootModule().Outputs) == 0 {
 		c.Ui.Error(fmt.Sprintf(
 			"The state file has no outputs defined. Define an output\n" +
 				"in your configuration with the `output` directive and re-run\n" +

--- a/command/output_test.go
+++ b/command/output_test.go
@@ -142,6 +142,27 @@ func TestOutput_noArgs(t *testing.T) {
 	}
 }
 
+func TestOutput_noState(t *testing.T) {
+	originalState := &terraform.State{}
+	statePath := testStateFile(t, originalState)
+
+	ui := new(cli.MockUi)
+	c := &OutputCommand{
+		Meta: Meta{
+			ContextOpts: testCtxConfig(testProvider()),
+			Ui:          ui,
+		},
+	}
+
+	args := []string{
+		"-state", statePath,
+		"foo",
+	}
+	if code := c.Run(args); code != 1 {
+		t.Fatalf("bad: \n%s", ui.ErrorWriter.String())
+	}
+}
+
 func TestOutput_noVars(t *testing.T) {
 	originalState := &terraform.State{
 		Modules: []*terraform.ModuleState{

